Added the form with php code to submit.php

GizKockesen opened the pull request:
So I've been going through tutorials and I think this works (hopefully).
I added the php code that should be able to add the necessary info
collected from the form to the Idea table in the database. Since I don't
have the info for connecting to the database, I didn't add the database
connection lines. I really hope this works!

GizKockesen committed:
Added the form with php code to submit.php

So I've been going through tutorials and I think this works (hopefully).
I added the php code that should be able to add the necessary info
collected from the form to the Idea table in the database. Since I don't
have the info for connecting to the database, I didn't add the database
connection lines. I really hope this works!

alex9311 committed:
added image uploading implementation

alex9311 commented:
yeah that would be great if you want to add it! 

I just did a pull request to merge my upload image example to your branch here: https://github.com/alex9311/TUD-Information-Retrieval-Group-02/pull/44

I thought it would be cool to get the full submission working on your branch then merge that into master once its done. What do you think?

GizKockesen commented:
I agree, on it! :)
On 13 Mar 2015 14:25, "Alex Simes" <notifications@github.com> wrote:

> yeah that would be great if you want to add it!
>
> I just did a pull request to merge my upload image example to your branch
> here: #44
> <https://github.com/alex9311/TUD-Information-Retrieval-Group-02/pull/44>
>
> I thought it would be cool to get the full submission working on your
> branch then merge that into master once its done. What do you think?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/alex9311/TUD-Information-Retrieval-Group-02/pull/43#issuecomment-78968899>
> .
>

GizKockesen committed:
Merge pull request #44 from alex9311/alex_upload_images

added image uploading implementation

alex9311 commented:
Great, let me know when when you're done! Then I can try to make it so we're storing the image's location on the server in the database as the image field.

GizKockesen commented:
Are we having two forms then?
On 13 Mar 2015 14:31, "Alex Simes" <notifications@github.com> wrote:

> Great, let me know when when you're done! Then I can try to make it so
> we're storing the image's location on the server in the database as the
> image field.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/alex9311/TUD-Information-Retrieval-Group-02/pull/43#issuecomment-78970906>
> .
>

GizKockesen commented:
Like nested, I mean
On 13 Mar 2015 14:36, "Gizem Kockesen" <gizemkockesen3@gmail.com> wrote:

> Are we having two forms then?
> On 13 Mar 2015 14:31, "Alex Simes" <notifications@github.com> wrote:
>
>> Great, let me know when when you're done! Then I can try to make it so
>> we're storing the image's location on the server in the database as the
>> image field.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/alex9311/TUD-Information-Retrieval-Group-02/pull/43#issuecomment-78970906>
>> .
>>
>

alex9311 commented on app/pages/file_upload.php:
```diff
@@ -0,0 +1,58 @@
+// if everything is ok, try to upload file
+} else {
+    if (move_uploaded_file($_FILES["fileToUpload"]["tmp_name"], $target_file)) {
+	session_start();
```
This is where we want to add the database entry stuff

alex9311 commented:
Ah yeah, I was thinking about it wrong actually.

See how my image uploader calls app/pages/file_upload.php when you press submit? That's submitting through a different file, like we were talking about before.

We want a single form, so only having to press submit once. It might be easiest to move your database login into app/pages/file_upload.php. Then we can upload the file and then make the entry into the database. That way we can get the file location from the file uploading logic (the $target_file variable) and store that into the database. 

Does that make sense? I made a comment in the code where I'm thinking the database stuff should go, if you look [here](https://github.com/alex9311/TUD-Information-Retrieval-Group-02/pull/43/files)

GizKockesen commented:
Should I move the database entry lines for text_description to file_upload.
php as well then? And also change the form action to file_upload. php?
On 13 Mar 2015 15:33, "Alex Simes" <notifications@github.com> wrote:

> Ah yeah, I was thinking about it wrong actually.
>
> See how my image uploader calls app/pages/file_upload.php when you press
> submit? That's submitting through a different file, like we were talking
> about before.
>
> We want a single form, so only having to press submit once. It might be
> easiest to move your database login into app/pages/file_upload.php. Then we
> can upload the file and then make the entry into the database. That way we
> can get the file location from the file uploading logic (the $target_file
> variable) and store that into the database.
>
> Does that make sense? I made a comment in the code where I'm thinking the
> database stuff should go, if you look here
> <https://github.com/alex9311/TUD-Information-Retrieval-Group-02/pull/43/files>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/alex9311/TUD-Information-Retrieval-Group-02/pull/43#issuecomment-78998448>
> .
>

alex9311 commented:
yeah that should work!

GizKockesen committed:
Updated submit.php and file_upload.php

Added Alex's image uploading code and moved the database connection and
entry to the Idea table to file_upload.php. Also changed the form action
to file_upload.php

GizKockesen commented:
Okay I updated the two files (submit and file_upload). Let me know if I
misunderstood something :)
On 13 Mar 2015 16:11, "Alex Simes" <notifications@github.com> wrote:

> yeah that should work!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/alex9311/TUD-Information-Retrieval-Group-02/pull/43#issuecomment-79029393>
> .
>

alex9311 committed:
trying to make upload_form.php more readable

alex9311 commented:
Thanks! Code looks good, I think you just had one small mistake. I made the fix in a separate branch and made a pull request to bring it into yours here: https://github.com/alex9311/TUD-Information-Retrieval-Group-02/pull/45

I also restructured the code a bit in file_upload.php because it was getting a bit cluttered.

Let me know what you think! I tested it and its working!

alex9311 committed:
adding some comments to file_upload.php

GizKockesen committed:
Merge pull request #45 from alex9311/alex-restructuring_upload_form.php

trying to make upload_form.php more readable

GizKockesen commented:
It's great! And yaayyy it's working :D
On 13 Mar 2015 18:15, "Alex Simes" <notifications@github.com> wrote:

> Thanks! Code looks good, I think you just had one small mistake. I made
> the fix in a separate branch and made a pull request to bring it into yours
> here: #45
> <https://github.com/alex9311/TUD-Information-Retrieval-Group-02/pull/45>
>
> I also restructured the code a bit in file_upload.php because it was
> getting a bit cluttered.
>
> Let me know what you think! I tested it and its working!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/alex9311/TUD-Information-Retrieval-Group-02/pull/43#issuecomment-79149113>
> .
>

alex9311 commented:
Great! Should we merge this into master now?

GizKockesen commented:
Yeah sure!
On 13 Mar 2015 19:02, "Alex Simes" <notifications@github.com> wrote:

> Great! Should we merge this into master now?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/alex9311/TUD-Information-Retrieval-Group-02/pull/43#issuecomment-79196287>
> .
>

alex9311 committed:
added page to view idea database

alex9311 commented:
I added a view_ideas.php page which prints the ideas in the database with images. I'm going to merge this branch in now!

GizKockesen commented:
awesome!
On 13 Mar 2015 19:32, "Alex Simes" <notifications@github.com> wrote:

> Merged #43
> <https://github.com/alex9311/TUD-Information-Retrieval-Group-02/pull/43>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/alex9311/TUD-Information-Retrieval-Group-02/pull/43#event-252387295>
> .
>
